### PR TITLE
Bug 1996156: show a default sidebar for resources that doesn't have sidebar

### PR DESCRIPTION
**Pull request by sahil143**
fixes: https://issues.redhat.com/browse/ODC-6261

Screenshot:
Screenshot 2021-08-19 at 10 24 30 PM

### CODE REVIEW

**Comment by invincibleJai** [comment]
/approve
 
look good needed to verify once.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]
@sahil143 Shouldn't the default resource details go into the details tab in the sidebar? The screenshot shows no tab in the sidebar.

**Comment by sahil143** [comment]
@rohitkrai03 This was the default behavior for a resource that doesn't have a sidebar which got removed while working on the topology workload panel. ResourceOverviewPage handles this itself we don't send any information about the tab or anything.

**Comment by rohitkrai03** [comment]
@sahil143 It feels odd to see it like that. I am okay if it was always like this but can we not put the resource overview into a details tab?

**Comment by sahil143** [comment]
@rohitkrai03 fixed
Screenshot 2021-08-20 at 5 18 46 PM

File: frontend/public/components/overview/resource-overview-page.tsx [hunk 1]
```diff
@@ -42,40 +44,53 @@ export const OverviewDetailsResourcesTab: React.SFC<OverviewDetailsResourcesTabP
...
+};
 
 export const ResourceOverviewPage = connectToModel(({ kindObj, item, customActions }) => {
   if (!kindObj && !item?.obj) {
     return null;
   }
   const resourceModel = kindObj || modelFor(referenceFor(item.obj));
   const ref = referenceForModel(resourceModel);
-  const loader = resourceOverviewPages.get(ref, () => Promise.resolve(DefaultOverviewPage));
+  const loader = resourceOverviewPages.get(ref, () => Promise.resolve(DefaultSideBar));
```
Comment (rohitkrai03) [inline frontend/public/components/overview/resource-overview-page.tsx:93]: Is this being used anywhere else apart from topology side bar? If yes do we want to change all of that? Or we should just add the tabs logic in topology sidebar only?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/public/components/overview/resource-overview-page.tsx (hunk 1 above)
Comment (sahil143) [reply frontend/public/components/overview/resource-overview-page.tsx:93]: This is not being used anywhere apart from Topology

File: frontend/public/components/overview/resource-overview-page.tsx [hunk 2]
```diff
@@ -42,40 +44,53 @@ export const OverviewDetailsResourcesTab: React.SFC<OverviewDetailsResourcesTabP
...
+  const menuActions = [
+    ...(customActions ? customActions : []),
+    ...Kebab.getExtensionsActionsForKind(kindObj),
+    ...common,
+  ];
+  return (
+    <ResourceOverviewDetails
+      item={item}
+      kindObj={kindObj}
+      menuActions={customActions ? [...customActions, ...menuActions] : menuActions}
```
Comment (rohitkrai03) [inline frontend/public/components/overview/resource-overview-page.tsx:81]: Do you need to check for `customActions` again here? Its already getting added in `menuActions` above.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**